internal/container/docker.go

```diff
@@ -392,13 +393,32 @@ func (e *dockerImpl) ImageList(ctx context.Context, references ...string) ([]Ima
 		return nil, nil
 	}
 
-	result := []Image{}
-	if err := json.Unmarshal(buf, &result); err != nil {
+	result, err := readJsonl[Image](bytes.NewBuffer(buf))
+	if err != nil {
 		return nil, err
 	}
+
 	return result, nil
 }
 
+func readJsonl[T any](r io.Reader) ([]T, error) {
```

Comment:
is it possible to add a unit test for this?

Reply:
added